Doc/library/shutil.rst

```diff
@@ -346,6 +346,8 @@ Directory and files operations
    .. versionchanged:: 3.13
       :func:`!rmtree` now ignores :exc:`FileNotFoundError` exceptions for all
       but the top-level path.
+      Exceptions other than :exc:`OSError` are now always propagated to
+      the caller.
```

Comment:
Your current wording is accurate, but maybe this would make it slightly clearer for readers? Not sure:

```suggestion
      Exceptions other than :exc:`OSError` and subclasses of :exc:`!OSError`
      are now always propagated to the caller.
```